app/models/course_data/articles_courses.rb

```diff
@@ -67,8 +67,13 @@ def update_cache
     # We use the 'all_revisions' scope so that the dashboard system edits that
     # create sandboxes are not excluded, since those are often wind up being the
     # first edit of a mainspace article's revision history
-    self.new_article = all_revisions.exists?(new_article: true)
-
+    article_revisions = article.revisions
+                               .where('date >= ?', course.start)
+                               .where('date <= ?', course.end)
+    article_created_at = article.created_at
```

Comment:
The `created_at` value for an `Article` is the timestamp of when the record in the Dashboard database got created, not when the actual Wikipedia page was created. A timestamp of a revision with `new_article: true` is the only marker we have for when an article was created.

I think the two conditions where we can mark it 'new' are a) the way we determined it previously, or b) if there exists a revision for that article that is both `system: true` and `new_article: true` and that revision happened during the course.

Reply:
Okay! So I completely removed the condition involving <code>article.created_at</code> as the work is itself going to be done by the revisions.

Reply:
So now it only checks the following conditions:
1. Course revision by students for that article with<code>new_article: true</code> (Earlier way)
2. Any article revision(by anyone, not necessarily students) during course dates having <code>system: true</code> and <code>new_article: true</code>